docs/tutorials/your-first-workflow.md

```diff
@@ -0,0 +1,345 @@
+        stop = {{inputs.parameters.stop}}
+        step = {{inputs.parameters.step}}
+
+        vals = np.arange(start,stop,step).tolist()
+        with open("/tmp/data.json", "w") as f:
+          json.dump(vals, f)
+    outputs:
+      artifacts:
+      - name: gridPoints
+        path: /tmp/data.json
+        archive:
+          none: { }
+```
+
+Now its time to plot a figure. This step will expect a list of gridPoints and compute the according function, then plot and save the figure. By default, the workflows engine will tarball outputs,
+so we configure `archive: { none: {} }` to prevent this, and give us a raw .png file.
+
+```yaml
+  - name: plot-the-figure
+    inputs:
+      parameters:
+        - name: gridPoints
+    script:
+      image: python:3.10
+      volumeMounts:
+      - name: tmpdir
+        mountPath: /tmp
+      command: [/tmp/venv/bin/python]
+      source: |
+        import matplotlib.pyplot as plt
+        from math import sin
+        import json
+
+        with open("/tmp/data.json", "r") as f:
+          x = json.load(f)
+        y = [sin(val) for val in x]
+        plt.plot(x,y)
+        plt.savefig("/tmp/output_fig.png")
+    outputs:
+      artifacts:
+      - name: sin-figure
+        path: "/tmp/output_fig.png"
+        archive:
+          none: { }
+```
+
+The bulk of the work is now done so we can now fill in the DAG from our original boilerplate. This workflow has three steps, with linear dependencies so arranging this shouldn't be difficult.
+Note the templating in the pre-processing step. By defining the parameter in the top level arguments, these can be easily overwritten when re-using the template. The respective arguments can
+be seen below.
+
+```yaml
+  - name: workflow-entry
+    dag:
+      tasks:
+      - name: install-dependencies
+        template: install-dependencies
+
+      - name: pre-processing
+        dependencies: [install-dependencies]
+        template: pre-processing
+        arguments:
+          parameters:
+          - name: start
+            value: "{{workflow.parameters.start}}"
+          - name: stop
+            value: "{{workflow.parameters.stop}}"
+          - name: step
+            value: "{{workflow.parameters.step}}"
+
+      - name: plot-figure
+        dependencies: [pre-processing]
+        template: plot-the-figure
+```
+
+```yaml
+spec:
+  entrypoint: workflow-entry
+  arguments:
+    parameters:
+      - name: start
+        value: "2"
+      - name: stop
+        value: "10"
+      - name: step
+        value: "5"
+```
+
+Now we have a whole workflow! After putting all the parts together and linting it, you can submit it to verify the behavior. Once validated, switch the `kind` back to `ClusterWorkflowTempalate`,
+and that should be finished!
+
+After loading the workflows module, you can submit the workflow with the `argo` tool
+
+```bash
+argo submit formatted-workflow.yaml -n <SESSION-ID> --server https://kubernetes.workflows.diamond.ac.uk
```

Comment:
this only works with ```kind: Workflow``` right? but just above you suggest to change back to ClusterWorkflowTemplate. Maybe it would make sense to explain in bit more detail that there are different ways of running a workflow:

1. from a Workflow manifest essentially with arguments hardcoded
2. from a Template - can the argo CLI submit from a template with arguments passed in at runtime? if yes, probably worth explaining
3. push template to one of the repos that is tracked and then submit via the web interface